### PR TITLE
systemtest: use Fleet Server

### LINKED ISSUE #5075: Update system tests to use Fleet Server
We'll need to update our system tests to use Fleet Server for 7.13. We can do this before or after feature freeze, depending on when Fleet server support is removed from Kibana.

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

In the Fleet integration system test, bootstrap Fleet Server and then run the APM integration on a separate Elastci Agent that enrolls with the Fleet Server.

## Related issues

Closes #5075

### CODE REVIEW

**Comment by axw** [comment]
test seems to be flaky, bringing back into draft while I debug...

File: systemtest/containers.go [hunk 1]
```diff
@@ -361,14 +370,27 @@ func (c *ElasticAgentContainer) Start() error {
 	defer cancel()
 
 	// Update request from user-definable fields.
-	if c.FleetEnrollmentToken != "" {
+	if c.FleetServer {
+		c.request.Env["FLEET_SERVER_ENABLE"] = "1"
+		c.request.Env["FLEET_SERVER_HOST"] = "0.0.0.0"
```
Comment (simitt) [inline systemtest/containers.go:375]: Shouldn't be necessary anymore afaik.

**Review by simitt — APPROVED** [review]
Looks great.

File: systemtest/containers.go (hunk 1 above)
Comment (axw) [reply systemtest/containers.go:375]: I just tried removing FLEET_SERVER_HOST and FLEET_SERVER_PORT (and tried just removing each individually), and the server never became accessible. In the elastic-agent log I see "State changed to CONFIG: Re-configuring". Maybe a bug masked by overriding the host/port?

Comment (axw) [reply systemtest/containers.go:375]: I'm going to merge as it is and come back to this, so we can get our system tests running again

Comment (simitt) [reply systemtest/containers.go:375]: Yes sure; thanks for trying it out.

**Pull request merged**